internal/pkg/services/user.go

```diff
@@ -103,3 +108,20 @@ func (u *UserService) UpdateUserPassword(email string, password string) error {
 
 	return nil
 }
+
+func (u *UserService) GetAllUsers() (*[]models.User, error) {
+	var users []models.User
+	rows, err := u.DB.Queryx(getAllUsers)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for rows.Next() {
+		var user models.User
+		_ = rows.StructScan(&user)
+		users = append(users, user)
+	}
```

Comment:
You can use the libs SliceScan() function to scan a slice of structs

Reply:
i tried to use that but i honestly could not figure out how to.